src/base/bittorrent/torrentinfo.cpp

```diff
@@ -213,24 +214,79 @@ QByteArray TorrentInfo::metadata() const
 
 QStringList TorrentInfo::filesForPiece(int pieceIndex) const
 {
-    if (pieceIndex < 0)
-        return QStringList();
+    // no checks here since fileIndicesForPiece() will return an empty list
+    QList<int> fileIndices = fileIndicesForPiece(pieceIndex);
+
+    QStringList res;
+    std::transform(fileIndices.begin(), fileIndices.end(), std::back_inserter(res), [this](int i) {
+        return this->filePath(i);
+    });
+
```

Comment:
Should we have 
```C++
template <class T>
A complexFunction(T v)
{
...
}
```
and call it like
```C++
int i; short s;
complexFunction(i);
complexFunction(s);
```
the binary size will increase comparing to

```C++
A complexFunction(int v)
{
...
}

int i; short s;
complexFunction(i);
complexFunction(s);
```

But here? Why?